app/models/api/messages/flowcell_io.rb

```diff
@@ -101,7 +101,9 @@ def detect_float_descriptor(name, ignored_unit)
 
         def detect_descriptor(name)
           # Sort here goes by id ascending, so we use a reverse each, in order to find the most recent descriptor with the passed in 'name'
-          lab_events.sort.reverse_each { |e| e.descriptor_value_for(name).tap { |bc| return bc if bc.present? } }
+          lab_events.sort.reverse_each do |e|
+            e.descriptor_value_for(name).tap { |value| return value if value.present? }
+          end
```

Comment:
This bit was picked up by @KatyTaylor in her code review on the last merge. Updated the confusingly named `bc` variable